src/main.ts

```diff
@@ -1,261 +1,3 @@
-    return {
-      any: [config]
-    };
-  }
-
-  return config;
-}
-
-function printPattern(matcher: IMinimatch): string {
-  return (matcher.negate ? "!" : "") + matcher.pattern;
-}
-
-function checkGlobs(
-  changedFiles: string[],
-  globs: StringOrMatchConfig[]
-): boolean {
-  for (const glob of globs) {
-    core.debug(` checking pattern ${JSON.stringify(glob)}`);
-    const matchConfig = toMatchConfig(glob);
-    if (checkMatch(changedFiles, matchConfig)) {
-      return true;
-    }
-  }
-  return false;
-}
-
-function isMatch(changedFile: string, matchers: IMinimatch[]): boolean {
-  core.debug(`    matching patterns against file ${changedFile}`);
-  for (const matcher of matchers) {
-    core.debug(`   - ${printPattern(matcher)}`);
-    if (!matcher.match(changedFile)) {
-      core.debug(`   ${printPattern(matcher)} did not match`);
-      return false;
-    }
-  }
-
-  core.debug(`   all patterns matched`);
-  return true;
-}
-
-// equivalent to "Array.some()" but expanded for debugging and clarity
-function checkAny(changedFiles: string[], globs: string[]): boolean {
-  const matchers = globs.map(g => new Minimatch(g));
-  core.debug(`  checking "any" patterns`);
-  for (const changedFile of changedFiles) {
-    if (isMatch(changedFile, matchers)) {
-      core.debug(`  "any" patterns matched against ${changedFile}`);
-      return true;
-    }
-  }
-
-  core.debug(`  "any" patterns did not match any files`);
-  return false;
-}
-
-// equivalent to "Array.every()" but expanded for debugging and clarity
-function checkAll(changedFiles: string[], globs: string[]): boolean {
-  const matchers = globs.map(g => new Minimatch(g));
-  core.debug(` checking "all" patterns`);
-  for (const changedFile of changedFiles) {
-    if (!isMatch(changedFile, matchers)) {
-      core.debug(`  "all" patterns did not match against ${changedFile}`);
-      return false;
-    }
-  }
-
-  core.debug(`  "all" patterns matched all files`);
-  return true;
-}
-
-function checkMatch(changedFiles: string[], matchConfig: MatchConfig): boolean {
-  if (matchConfig.all !== undefined) {
-    if (!checkAll(changedFiles, matchConfig.all)) {
-      return false;
-    }
-  }
-
-  if (matchConfig.any !== undefined) {
-    if (!checkAny(changedFiles, matchConfig.any)) {
-      return false;
-    }
-  }
-
-  return true;
-}
-
-async function addLabels(
-  client: github.GitHub,
-  prNumber: number,
-  labels: string[]
-) {
-  await client.issues.addLabels({
-    owner: github.context.repo.owner,
-    repo: github.context.repo.repo,
-    issue_number: prNumber,
-    labels: labels
-  });
-}
-
-async function removeLabels(
-  client: github.GitHub,
-  prNumber: number,
-  labels: string[]
-) {
-  await Promise.all(
-    labels.map(label =>
-      client.issues.removeLabel({
-        owner: github.context.repo.owner,
-        repo: github.context.repo.repo,
-        issue_number: prNumber,
-        name: label
-      })
-    )
-  );
-}
+import { run } from "./labeler";
```

Comment:
do we have to make this change? anything changed between `labeler.ts` and `main.ts`?

Reply:
i am fine with changing it, just want to make sure there are no functional changes since the diff is not really helpful. 😄

Reply:
> anything changed between labeler.ts and main.ts?

No changes. I just moved everything from main into labeler to avoid invoking `run()` at the end of the file, so we can test it in isolation. The diff makes it look like a bigger change than it is :)